packages/fxa-settings/src/lib/model-data/bind-decorator.ts

```diff
@@ -135,3 +146,94 @@ export class InvalidModelInstance extends Error {
     );
   }
 }
+function writeRawValue(value: any, dataType: dataType | undefined) {
+  if (dataType !== undefined && typeof value !== dataType) {
+    throw new Error(
+      `Data type mismatch! Type safety violated! Cannot assign ${typeof value} to ${dataType}.`
+    );
+  }
+  if (dataType === 'object') {
+    return JSON.stringify(value);
+  }
+  return value.toString();
+}
+
+/**
+ * Takes a string value and tries to convert it to the specified data type.
+ * @param value - raw string value
+ * @param dataType - expected type
+ * @returns - a value of the specified type
+ */
+function coerceValue(
+  value: string | undefined,
+  dataType: dataType | undefined
+): string | boolean | number | any | null {
+  if (value == null) {
+    return value;
+  }
+
+  if (typeof value === 'string') {
+    if (dataType === 'string') {
+      return value;
+    }
+
+    if (dataType === 'number') {
+      return parseFloat(value);
+    }
+
+    if (dataType === 'boolean') {
+      const lower = value.toLowerCase().trim();
+      if (lower !== 'true' && lower !== 'false') {
+        throw new Error('Invalid boolean value: ' + value);
+      }
+      return value === 'true';
+    }
+
+    // Value is a string with unknown data type or data type of object
+    try {
+      return JSON.parse(value);
+    } catch {
+      console.warn(`Cannot JSON parse value! ${value}`);
```

Comment:
Not sure if we want to leave this here, but it is definitely useful if something unexpected happens.

Reply:
I don't mind it.  But we would only see it if a) we notice it in our own browsers, or b) it happens to be in the console as part of Sentry's breadcrumbs for some other error.  So if you want to know about these errors I think you need to re-throw or explicitly capture them with Sentry.

Reply:
Sounds like good advice to me. Thanks y'all.